mimic/test/test_swift.py

```diff
@@ -119,74 +313,617 @@ def test_get_container(self):
                 b"X-Container-Bytes-Used")[0], b"0"
         )
 
-    def test_get_no_container(self):
+    def test_get_container_non_existent(self):
         """
         GETing a container that has not been created results in a 404.
         """
-        self.createSwiftService()
         # create a container
-        uri = (self.json_body['access']['serviceCatalog'][0]['endpoints'][0]
-               ['publicURL'] + '/testcontainer').encode("ascii")
-        container_response = self.successResultOf(
-            request(self, self.root, b"GET", uri)
+        container_response, _ = self.get_container(
+            expected_result=404, with_body=False)
+
+        # Validate the response
+        self.assertIsNone(
+            container_response.headers.getRawHeaders(
+                "X-Container-Object-Count")
         )
-        self.assertEqual(container_response.code, 404)
+        self.assertIsNone(
+            container_response.headers.getRawHeaders(
+                "X-Container-Bytes-Used")
+        )
+
+    def test_get_container_non_empty(self):
+        """
+        Test retrieving a non-empty container
+        """
+        # create a container
+        self.put_container()
+
+        # put the object
+        self.put_object(headers={b"content-type": [b"text/plain"]})
+
+        # get the container
+        container_response, container_contents = self.get_container()
+
+        # validate the response
+        self.assertNotEqual(container_contents, [])
+        self.assertEqual(len(container_contents), 1)
+        self.assertEqual(container_contents[0]['name'],
+                         self.object_path.decode('utf-8'))
+        self.assertEqual(container_contents[0]['content_type'], "text/plain")
+        self.assertEqual(container_contents[0]['bytes'], self.object_size)
+        self.assertEqual(
+            container_response.headers.getRawHeaders(
+                b"X-Container-Object-Count")[0], b"1"
+        )
+        self.assertEqual(
+            container_response.headers.getRawHeaders(
+                b"X-Container-Bytes-Used")[0],
+            "{0}".format(self.object_size).encode("utf-8")
+        )
+
+    def test_head_container(self):
+        """
+        Validate the HEAD operation on a container.
+        """
+        # create a container
+        self.put_container()
+
+        # the container should have some information in it
+        # the names of the object's don't matter, but having some
+        # dynamic content will help prove it's all working correctly
+        # so generate a series of objects names to use
+        object_paths = [
+            self.object_path,
+        ]
+        object_path_prefix = b""
+        for i in range(10):
```

Comment:
Control structures (loops, conditionals, etc) in tests are generally a bad idea, because if the test starts failing, this makes it much harder to quickly see what exact property was failing to verify.  Also, control structures are logic, and logic can have bugs, much more readily than a linear set of constants can have bugs.  However, I can see why you would want to have a bit of generative data here to ensure the various data structures you're populating are not interfering with each other, and so on.  Just something to keep in mind for the future.